### DISCUSSION TITLE
1.15.2 - User Context

### DISCUSSION THREAD

**Discussion by rvtdadmin** [Romanitho/Winget-AutoUpdate#210] [General]
- If no user is logged in the log just hangs at "Starting WAU in User context". It seems to run again the next time the Scheduled Task is triggered. Not sure what state this leaves the WAU script process at.

- I'm having issues with users getting UAC prompts when the user context runs. I'm sure this is because some programs are installing something in the user context and winget sees that but they are not actually user context installs. Not sure what can be done about this as it might be an issue with how the program is installed and how winget detects them.
These are a couple that did that. 
-> Available update : PDFCreator. 
-> Available update : Microsoft Visual C++ 2015-2019 Redistributable (x86)

- In the case above if a user is getting a UAC prompt and their computer is locked. WAU hangs waiting for input in the UAC prompt.

This is of course causing a lot of calls to the helpdesk so I might not be able to use the user-context part of WAU.
What is the best way to disable the user context autoupdate? Just disabling the Winget-AutoUpdate-UserContext Scheduled task causes it to hang in the log at "Starting WAU in User context"

Thanks!

**Comment by Romanitho** [comment]
Thanks for detailed feedback. I would recommend to install WAU with `-DisableWAUAutoUpdate` in corporate environment to avoid big issues as we do not have a lot of testers (yet).
Obviously we need to implement an option to choose to have the User context feature or not. We didn't measure the impact as most of apps should be installed under system.

**Reply by Romanitho** [reply]
To stop user context, just remove the `Winget-AutoUpdate-UserContext` scheduled task. This is an option we need to add.

**Reply by KnifMelti** [reply]
> Thanks for detailed feedback. I would recommend to install WAU with `-DisableWAUAutoUpdate` in corporate environment to avoid big issues as we do not have a lot of testers (yet). Obviously we need to implement an option to choose to have the User context feature or not. We didn't measure the impact as most of apps should be installed under system.

"As is"...
...that's why I'm adding the noWAUupdate in my environment. 😘 (and rerunning the installation when updating)
Would be nice to have a few more testers... 💕💕💕💕

**Comment by KnifMelti** [comment]
In my experience you'll have to exclude things like Visual C++ (or run a whitelist) because of all the strange things happening in Winget land...
...had a .Net installation triggering UAC in user mode because of an old installation was detected in the ARP (but it didn't exist)...

**Comment by rvtdadmin** [comment]
Thanks!

I am updating the installs to disableautoupdate and remove usercontext scheduled task for now.

**Reply by Romanitho** [reply]
as @KnifMelti said, you might keep AutoUpdate feature ON and use white listing to manage the entire process of update, with few more testings on your side.
But to be honest, what happened to this last version is my fault, we changed the behavior of WAU. We used to add new features only on "fresh" install (or on reinstall). So we will as much as possible continue to implement features that can cause minimum of side effects.

**Reply by rvtdadmin** [reply]
All good, appreciate your work on WAU!
I may enable auto-update again later, thanks for the info on how you roll out features.
